src/NServiceBus.Core/Routing/RoutingFeature.cs

```diff
@@ -47,7 +47,7 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 EnableBestPracticeEnforcement(context);
             }
 
-            unicastBusConfig?.MessageEndpointMappings.Apply(publishers, unicastRoutingTable, transportInfrastructure.MakeCanonicalForm, conventions);
+            unicastBusConfig?.MessageEndpointMappings.Apply(publishers, unicastRoutingTable, transportInfrastructure.MakeCanonicalForm, conventions, context.Settings.GetAvailableTypes());
```

Comment:
With unobtrusive mode `GetAvailableTypes` might not contain all the types you need. Are we sure this is correct?